From 45c42df5f01044b87c01b53d47321c5c80da47bf Mon Sep 17 00:00:00 2001
From: Raphael Collet <rco@openerp.com>
Date: Mon, 12 Dec 2016 17:14:27 +0100
Subject: [PATCH] [FIX] stock: remove `_parent_store=True` from model
 `stock.quant.package`

The model does not actually need this feature, as the hierarchies of packages
are not expected to be deep.  Performance wise, updating columns `parent_left`
and `parent_right` may actually suffer from concurrent updates that require
retrying requests.

Backport of 6fa21672470b46d8d169bdf9d8955813cf8ddeeb

Fixes #25459
---
 addons/stock/models/stock_quant.py | 7 +------
 1 file changed, 1 insertion(+), 6 deletions(-)

diff --git a/addons/stock/models/stock_quant.py b/addons/stock/models/stock_quant.py
index 6006d35..aa4f3d7 100644
--- a/addons/stock/models/stock_quant.py
+++ b/addons/stock/models/stock_quant.py
@@ -579,10 +579,7 @@ class QuantPackage(models.Model):
     """ Packages containing quants and/or other packages """
     _name = "stock.quant.package"
     _description = "Physical Packages"
-    _parent_name = "parent_id"
-    _parent_store = True
-    _parent_order = 'name'
-    _order = 'parent_left'
+    _order = 'name'
 
     name = fields.Char(
         'Package Reference', copy=False, index=True,
@@ -595,8 +592,6 @@ class QuantPackage(models.Model):
     ancestor_ids = fields.One2many('stock.quant.package', string='Ancestors', compute='_compute_ancestor_ids')
     children_quant_ids = fields.One2many('stock.quant', string='All Bulk Content', compute='_compute_children_quant_ids')
     children_ids = fields.One2many('stock.quant.package', 'parent_id', 'Contained Packages', readonly=True)
-    parent_left = fields.Integer('Left Parent', index=True)
-    parent_right = fields.Integer('Right Parent', index=True)
     packaging_id = fields.Many2one(
         'product.packaging', 'Package Type', index=True,
         help="This field should be completed only if everything inside the package share the same product, otherwise it doesn't really makes sense.")
-- 
2.7.4

